### PR TITLE
Add i3c-core RTL code

### PR DESCRIPTION
**Pull request by ArthurHeymans**

### CODE REVIEW

**Review by swenson — DISMISSED** [review]
LGTM, though I'd prefer it were loaded somewhere else in memory that is closer to our other peripherals (like in the `0x200x` range) so that we leave a cleaner footprint for integrators.

**Comment by ArthurHeymans** [comment]
> LGTM, though I'd prefer it were loaded somewhere else in memory that is closer to our other peripherals (like in the `0x200x` range) so that we leave a cleaner footprint for integrators.

SW-emulator sets it to 0x1003_8000. I can update that.

File: registers/bin/generator/src/main.rs [hunk 1]
```diff
@@ -41,6 +41,8 @@ static CALIPTRA_RDL_FILES: &[&str] = &[
     "src/integration/rtl/caliptra_reg.rdl",
 ];
 
+static I3C_CORE_RDL_FILES: &[&str] = &["src/rdl/registers.rdl"];
```
Comment (jhand2) [inline registers/bin/generator/src/main.rs:44]: I was under the impression that Caliptra Core would just see the recovery interface and not bus-specific details. Could you point me at any spec for what this i3c RTL in Caliptra Core is for?

Comment (ArthurHeymans) [reply registers/bin/generator/src/main.rs:44]: So the recovery interface is part of the i3c periph. The intention is indeed to only use that bus abstracted part.
SecFwRecoveryIf in https://github.com/chipsalliance/i3c-core/tree/main/src/rdl is that interface we'll be targetting.

I think the other capabilities like TTI (I3C Target) will be used by MCU.

**Review by mhatrevi — DISMISSED** [review]

File: registers/bin/generator/src/main.rs (hunk 1 above)
Comment (jhand2) [reply registers/bin/generator/src/main.rs:44]: Ok, thanks for the details. @bharatpillilli would it make sense to factor the recovery interface RTL to be independent from the i3c peripheral RTL?

Not a huge deal, just thinking it makes the organization more clear for what Caliptra Core depends on.

**Review by jhand2 — DISMISSED** [review]

**Review by swenson — APPROVED** [review]

**Pull request merged**